modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotCheckProcess.java

```diff
@@ -0,0 +1,549 @@
+            }
+
+            return new GridFinishedFuture<>();
+        }
+
+        GridFutureAdapter<CheckResultDTO> locMetasChkFut = new GridFutureAdapter<>();
+
+        stopFutureOnAnyFailure(locMetasChkFut, () -> {
+            assert locReq.fut() == null;
+
+            locReq.fut(locMetasChkFut);
+
+            IgniteSnapshotManager snpMgr = kctx.cache().context().snapshotMgr();
+
+            if (log.isDebugEnabled())
+                log.debug("Checking local snapshot metadatas. Request=" + locReq + '.');
+
+            Collection<Integer> grpIds = F.isEmpty(locReq.groups()) ? null : F.viewReadOnly(locReq.groups(), CU::cacheId);
+
+            // An error can occur when the local future is not initialized yet.
+            Throwable locReqErr = locReq.error();
+
+            if (locReqErr != null)
+                locMetasChkFut.onDone(locReqErr);
+
+            if (!locMetasChkFut.isDone()) {
+                snpMgr.checker().checkLocalMetas(
+                    snpMgr.snapshotLocalDir(locReq.snapshotName(), locReq.snapshotPath()),
+                    grpIds,
+                    kctx.cluster().get().localNode().consistentId()
+                ).whenComplete((locMetas, err) -> {
+                    if (err != null)
+                        locMetasChkFut.onDone(err);
+                    else {
+                        if (!F.isEmpty(locMetas))
+                            locReq.meta(locMetas.get(0));
+
+                        locMetasChkFut.onDone(new CheckResultDTO(locMetas));
+                    }
+                });
+            }
+        });
+
+        return locMetasChkFut;
+    }
+
+    /** Phase 1 end. */
+    private void reducePreparationAndMetasCheck(
+        UUID procId,
+        Map<UUID, CheckResultDTO> results,
+        Map<UUID, Throwable> errors
+    ) {
+        SnapshotCheckProcessRequest locReq = currentRequest(snpName(results), procId);
+
+        assert locReq == null || locReq.opCoordId != null;
+
+        if (locReq == null || stopAndCleanOnError(locReq, errors) || !kctx.localNodeId().equals(locReq.opCoordId))
+            return;
+
+        Throwable stopClusterProcErr = null;
+
+        try {
+            assert !F.isEmpty(results) || !F.isEmpty(errors) || locReq.error() != null;
+
+            if (locReq.error() != null)
+                throw locReq.error();
+
+            locReq.metas = new HashMap<>();
+
+            Map<ClusterNode, Exception> errs = new HashMap<>();
+
+            results.forEach((nodeId, nodeRes) -> {
+                // A node might be non-baseline (not required).
+                if (locReq.nodes.contains(nodeId)) {
+                    assert nodeRes != null && nodeRes.partsHashes == null;
+                    assert kctx.cluster().get().node(nodeId) != null;
+
+                    locReq.metas.put(kctx.cluster().get().node(nodeId), nodeRes.metas);
+                }
+            });
+
+            errors.forEach((nodeId, nodeErr) -> {
```

Comment:
If `errors` are not empty this code is never executed. It is handled within `stopAndCleanOnError`